packages/tadviewer/src/components/DataGrid.tsx

```diff
@@ -361,6 +360,21 @@ const createGrid = (
   const selectionModel = new CellSelectionModel();
   grid.setSelectionModel(selectionModel);
   selectionModel.onSelectedRangesChanged.subscribe((e: any, args: any) => {
+    const { fromCell, toCell, fromRow, toRow } = args[0];
+    // if user moved to a single cell using arrow keys
+    if (fromCell === toCell && fromRow === toRow) {
```

Comment:
It looks like this will only invoke the `onGridSelectionChange` callback in the special case that the selection is just a single cell.  What happens when the user makes a rectangular selection?  What if they do so after making a single cell selection?  It looks like the callback won't be invoked in that case.

Reply:
Correct, this PR is only handling navigating around the grid via arrow keys - multi-select will come later. No rectangular behavior has changed. ([Linear link](https://linear.app/mother-duck/issue/SURF-1297/summary-for-multiple-selected-cells-slickgridtad))